### PR TITLE
extract native-token crate

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
`solana_program::native_token` imposes a `solana_program` requirement on `solana_svm` and also blocks moving out `solana_sdk::fee` 

#### Summary of Changes
Move to its own crate and re-export for backwards compatibility

### CODE REVIEW

File: sdk/program/src/lib.rs [hunk 1]
```diff
@@ -183,7 +183,6 @@
 //! [ed25519]: https://ed25519.cr.yp.to/
 //! [`Keypair`]: https://docs.rs/solana-sdk/latest/solana_sdk/signer/keypair/struct.Keypair.html
 //! [SHA-256]: https://en.wikipedia.org/wiki/SHA-2
-//! [`Sol`]: native_token::Sol
```
Comment (joncinque) [inline sdk/program/src/lib.rs]: Shouldn't this still work thanks to the re-export? If not, we should remove all of the other links here too (can be done in another PR though)

**Review by joncinque — COMMENTED** [review]
Looks great, thanks! @yihau, can you accept ownership of `solana-native-token` once it's passed over?

File: sdk/program/src/lib.rs (hunk 1 above)
Comment (kevinheavey) [reply sdk/program/src/lib.rs]: yeah i don't remember why that line got deleted. Adding it back

**Review by joncinque — APPROVED** [review]
Looks great!

**Pull request merged**